js/counties.js

```diff
@@ -0,0 +1,73 @@
+function hideCounties() {
+  map.selectAll('.county')
+    .data([])
+    .exit()
+    .remove();
+}
+
+function showCounties(state) {
+  loadCountyData(state, displayCountyData);
+}
+
+function loadCountyData(state, callback) {
+  // if the county boundaries for this state are already loaded, do nothing.
+  // otherwise load them now
+  if(!countyData.has(state)) {
+    // convert state (an ID) into a file name (FIPS-based, at least for now)
+    var stateFIPS = stateDict.filter(function(d) {
+      return(d.state_abbv === state);
+    })[0].state_FIPS;
+    
+    // download the data and run the processing function
+    oneStateCounties = d3.json("data/" + stateFIPS + "-quantized.json", function(error, oneStateCounties) {
+      if(error) callback(error);
+      countyData.set(state, oneStateCounties);
+      callback(error, countyData.get(state));
+    });
+  } else {
+    // if we already have the data, just run the processing function
+    callback(null, countyData.get(state));
```

Comment:
hmm. could do that...i used `callback` because it's a fairly common argument name (e.g., is the name of the 2nd arg to `d3.json`, https://github.com/d3/d3-request#json) and makes sense to me because it's the function that gets called after the data loading happens, just as in d3.json. so `callback` feels descriptive to me. but i'm open to something else - what would be clearer to you?

Reply:
Well, this might just be a convention I need to learn! Keep it because it sounds like changing it might not be beneficial to anyone since it's a standard practice.

Reply:
OK, keeping it for now. Happy to discuss again later if it continues to not work for you. Even though `callback` isn't a keyword, the whole `callback` idea is pretty specific to javascript because it's tied to the asynchronous execution thing. Over time I hope we'll continue to become more comfortable with it...and maybe also find better ways to name our callback arguments in the process =).